analytics-core/src/main/scala/org/ekstep/analytics/framework/DataFetcher.scala

```diff
@@ -71,4 +76,21 @@ object DataFetcher {
         null;
     }
 
+    def getFilteredKeys(query: Query, keys: Array[String], partitions: Option[List[Int]]): Array[String] = {
+        if (partitions.nonEmpty) {
+            val dates = CommonUtil.getQueryDates(query)
```

Comment:
Dates check is redundant. Can we just do regex match for the position 11,12,13 index position?

Reply:
Done - updated to use regex match for {date}-{partition}